tests/integration/dynamic_ip_test.py

```diff
@@ -1008,3 +1011,83 @@ def test_dummy_existance_after_ipv6_autoconf_timeout(dummy00):
     # According to RFC 4861, autoconf(IPv6-RA) will instruct client to do
     # DHCPv6 or not. With autoconf timeout, DHCPv6 will not start.
     assertlib.assert_state({Interface.KEY: [ifstate]})
+
+
+def test_static_ip_with_routes_switch_back_to_dynamic(dhcpcli_up):
```

Comment:
The static IP with its routes fit a fixture as they are not part of the test itself but as its setup.
But I leave it up to you, this is not a must as we do not enforce it well enough anyway.